Sources/SourceKitLSP/Swift/SwiftLanguageService.swift

```diff
@@ -815,15 +831,31 @@ extension SwiftLanguageService {
       additionalParameters: additionalCursorInfoParameters
     )
 
-    return cursorInfoResponse.refactorActions.compactMap {
-      do {
-        let lspCommand = try $0.asCommand()
-        return CodeAction(title: $0.title, kind: .refactor, command: lspCommand)
-      } catch {
-        logger.log("Failed to convert SwiftCommand to Command type: \(error.forLogging)")
-        return nil
+    var canInlineMacro = false
+
+    let showMacroExpansionsIsEnabled = await self.sourceKitLSPServer?.options.experimentalFeatures.contains(
+      .showMacroExpansions
+    )
+
+    var refactorActions = cursorInfoResponse.refactorActions.compactMap {
+      let lspCommand = $0.asCommand()
+      canInlineMacro = $0.actionString == "source.refactoring.kind.inline.macro"
+
+      if !(showMacroExpansionsIsEnabled != nil && showMacroExpansionsIsEnabled!) {
+        canInlineMacro = false
       }
```

Comment:
Do you think this would read cleaner as follows? I found the setting of `canInlineMacro` to `true` just to revert it back to `false` a little confusing. I also noticed that we would set `canInlineMacro` back to `false` if `source.refactoring.kind.inline.macro` was not the last refactoring action.

```suggestion
      if !canInlineMacro, showMacroExpansionsIsEnabled {
        canInlineMacro = $0.actionString == "source.refactoring.kind.inline.macro"
      }
```